src/Microsoft.AspNet.Authentication/AuthenticationOptions.cs

```diff
@@ -42,5 +43,11 @@ public string AuthenticationScheme
         /// Additional information about the authentication type which is made available to the application.
         /// </summary>
         public AuthenticationDescription Description { get; set; } = new AuthenticationDescription();
+
+        /// <summary>
+        /// The request path within the application's base path where the user-agent will be returned
+        /// when a failure occurs.
+        /// </summary>
+        public PathString ErrorHandlerPath { get; set; }
```

Comment:
The question is: do we want to have that in the base handler? This error handling is only applicable to authentication handlers implementing `InvokeAsync` (since you can't control the flow elsewhere) and I'm not sure all middleware will implement it.